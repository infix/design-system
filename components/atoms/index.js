import Button from './Button'
import Input from './Input'
import Heading from './Heading'
import Text from './Text'
import IconButton from './IconButton'
import { RadioGroup, RadioButton } from './Radio'
import { Select } from './Select'

import { TwitterShareButton, FacebookShareButton } from './ShareButton'

const atoms = {
  Text,
  Heading,
  Button,
  IconButton,
  Input,
  RadioGroup,
  RadioButton,
  Select,
  TwitterShareButton,
  FacebookShareButton
}

export default atoms
