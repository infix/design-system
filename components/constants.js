/* global module */
const defaultWeights = [
  400,
  700
]

const defaultBreakpoints = [
  40,
  52,
  64
]

const defaultSpace = [
  0,
  8,
  16,
  32,
  64
]

const defaultFontSizes = [
  12,
  14,
  16,
  20,
  24,
  32,
  48,
  64,
  72
]

module.exports = {
  defaultBreakpoints,
  defaultSpace,
  defaultFontSizes,
  defaultWeights
}
