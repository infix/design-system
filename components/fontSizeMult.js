/* global require */
import {
  style,
  px
} from 'styled-system'

const {
  defaultFontSizes
} = require('./constants')

export const fontSizeMult = (propName, k) => {
  /* eslint-disable no-console */
  console.error('usage of fontSizeMult will be deprecated')
  /* eslint-enable no-console */

  return style({
    prop: 'fontSize',
    cssProperty: propName,
    key: 'fontSizes',
    transformValue: n => px(n * k),
    scale: defaultFontSizes
  })
}

export default fontSizeMult
