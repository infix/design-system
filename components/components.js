import atoms from './atoms'
import molecules from './molecules'
import organisms from './organisms'

import BarChart from './charts/BarChart'
import PieChart from './charts/PieChart'

/* OONI produced atoms */

export const Button = atoms.Button
export const Input = atoms.Input
export const RadioButton = atoms.RadioButton
export const RadioGroup = atoms.RadioGroup
export const TwitterShareButton = atoms.TwitterShareButton
export const FacebookShareButton = atoms.FacebookShareButton
export const Heading = atoms.Heading
export const Text = atoms.Text
export const Select = atoms.Select

/* OONI produced molecules */
export const InputWithIconButton = molecules.InputWithIconButton
export const LogoOONIRun = molecules.LogoOONIRun
export const Card = molecules.Card

/* OONI produced organisms */
export const Hero = organisms.Hero
export const HeroLead = organisms.HeroLead
export const Modal = organisms.Modal

/* Here we just export rebass until we implement all of these */
import {
  Absolute,
  Arrow,
  Avatar,
  BackgroundImage,
  Badge,
  Banner,
  BlockLink,
  Blockquote,
  Border,
  ButtonCircle,
  ButtonOutline,
  ButtonTransparent,
  Carousel,
  CarouselSlide,
  Checkbox,
  Circle,
  Close,
  Code,
  Column,
  Container,
  Divider,
  Donut,
  DotButton,
  Drawer,
  Embed,
  Fixed,
  Group,
  Image,
  Label,
  Lead,
  Link,
  Measure,
  Media,
  Message,
  NavLink,
  Overlay,
  Panel,
  PanelFooter,
  PanelHeader,
  Pre,
  Progress,
  Radio,
  Relative,
  Row,
  Samp,
  ScrollCarousel,
  Slider,
  Small,
  Star,
  Sticky,
  Subhead,
  Switch,
  TabItem,
  Tabs,
  Textarea,
  Toolbar,
  Tooltip,
  Truncate
} from 'rebass'

export { default as BarChart } from './charts/BarChart'
export { default as PieChart } from './charts/PieChart'

const components = {
  Button,
  Heading,
  Text,
  Input,
  Select,
  Card,
  RadioButton,
  RadioGroup,
  InputWithIconButton,
  HeroLead,
  Hero,
  LogoOONIRun,
  TwitterShareButton,
  FacebookShareButton,
  Modal,
  BarChart,
  PieChart,
  Absolute,
  Arrow,
  Avatar,
  BackgroundImage,
  Badge,
  Banner,
  BlockLink,
  Blockquote,
  Border,
  ButtonCircle,
  ButtonOutline,
  ButtonTransparent,
  Carousel,
  CarouselSlide,
  Checkbox,
  Circle,
  Close,
  Code,
  Column,
  Container,
  Divider,
  Donut,
  DotButton,
  Drawer,
  Embed,
  Fixed,
  Group,
  Image,
  Label,
  Lead,
  Link,
  Measure,
  Media,
  Message,
  NavLink,
  Overlay,
  Panel,
  PanelFooter,
  PanelHeader,
  Pre,
  Progress,
  Radio,
  Relative,
  Row,
  Samp,
  ScrollCarousel,
  Slider,
  Small,
  Star,
  Sticky,
  Subhead,
  Switch,
  TabItem,
  Tabs,
  Textarea,
  Toolbar,
  Tooltip,
  Truncate
}

export default components
