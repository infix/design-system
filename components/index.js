import {
  Provider,
  Flex,
  Box
} from 'rebass'

import colors from './theme/colors'
import theme from './theme/rebassTheme'

export { default as colors } from './theme/colors'
export { default as theme } from './theme/rebassTheme'

import components from './components'

// We export these from rebass directly
export {
  Provider,
  Flex,
  Box
} from 'rebass'

export const {
  Button,
  ButtonOutline,
  ButtonCircle,
  ButtonTransparent,
  Link,
  NavLink,
  BlockLink,
  Heading,
  Subhead,
  Text,
  Small,
  Lead,
  Pre,
  Code,
  Samp,
  Blockquote,
  Measure,
  Truncate,
  Label,
  Input,
  Select,
  Textarea,
  Checkbox,
  Radio,
  Slider,
  Image,
  Avatar,
  BackgroundImage,
  Container,
  Divider,
  Border,
  Media,
  Card,
  Banner,
  Panel,
  PanelHeader,
  PanelFooter,
  Progress,
  Message,
  Group,
  Toolbar,
  Badge,
  Circle,
  Tabs,
  TabItem,
  DotButton,
  Close,
  Relative,
  Absolute,
  Fixed,
  Sticky,
  Drawer,
  Overlay,
  Carousel,
  ScrollCarousel,
  CarouselSlide,
  Tooltip,
  Switch,
  Arrow,
  Star,
  Embed,
  Donut,
  Row,
  Column,
  RadioGroup,
  RadioButton,
  HeroLead,
  Hero,
  LogoOONIRun,
  InputWithIconButton,
  TwitterShareButton,
  FacebookShareButton,
  Modal,
  BarChart,
  PieChart
} = components

export const OONIComponents = Object.assign({}, components, { Provider, Box, Flex, colors, theme })

export default OONIComponents
