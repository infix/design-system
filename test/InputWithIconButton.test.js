/* global describe, test, expect, jest */
import React from 'react'
import { renderJSON, shallowWithTheme } from './index'
import { InputWithIconButton } from '../components'

describe('InputWithIconButton', () => {
  test('renders', () => {
    const json = renderJSON(
      <InputWithIconButton icon={<div />} />
    )
    expect(json).toMatchSnapshot()
  })

  test('render with onAction prop', () => {
    const json = renderJSON(
      <InputWithIconButton onAction={() => {}} icon={<div />} />
    )
    expect(json).toMatchSnapshot()
  })

  test('onAction is called on click', () => {
    const onAction = jest.fn()
    const wrapper = shallowWithTheme(
      <InputWithIconButton onAction={onAction} />
    )
    wrapper.find('IconButton').simulate('click')
    expect(onAction).toHaveBeenCalled()
  })
})
