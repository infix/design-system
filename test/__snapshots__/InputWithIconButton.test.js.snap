// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputWithIconButton render with onAction prop 1`] = `
.c1 {
  box-sizing: border-box;
}

.c2 {
  box-sizing: border-box;
  margin-top: 8px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c6 {
  border-radius: 56px;
  display: inline-block;
  height: 70px;
  line-height: 1;
  padding-left: 56px;
  padding-right: 56px;
  text-transform: uppercase;
  vertical-align: middle;
  -webkit-tap-highlight-color: transparent;
  outline: 0;
  font-family: inherit;
  font-weight: 600;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #f8f9fa;
  background-color: #0588cb;
  border: none;
  text-align: center;
  -webkit-letter-spacing: .5px;
  -moz-letter-spacing: .5px;
  -ms-letter-spacing: .5px;
  letter-spacing: .5px;
  z-index: 1;
  -webkit-transition: .2s ease-out;
  transition: .2s ease-out;
  cursor: pointer;
}

.c6:hover {
  border: none;
  background-color: #37a6ed;
}

.c6:active {
  -webkit-transition: .2s ease-in;
  transition: .2s ease-in;
  border: none;
  background-color: #0f77b8;
}

.c5 {
  padding-left: 16px;
  padding-right: 16px;
  font-size: 28px;
}

.c3 {
  font-size: 16px;
  font-family: inherit;
  display: 'block';
  vertical-align: 'middle';
  background-color: transparent;
  border: none;
  border-bottom: 1px solid #f1f3f5;
  border-radius: 0;
  margin: 0;
  padding: 0;
  outline: none;
  width: 100%;
  height: 32px;
  box-shadow: none;
  box-sizing: content-box;
  -webkit-transition: all .3s;
  transition: all .3s;
}

.c3:hover:not([readonly]) {
  border-bottom: 1px solid #dee2e6;
}

.c3:focus:not([readonly]) {
  border-bottom: 2px solid #0588cb;
}

.c4 {
  padding-left: 16px;
  padding-right: 16px;
  font-size: 28px;
  border-radius: none;
  color: #adb5bd;
  background-color: transparent;
}

.c4:hover {
  background-color: transparent;
  color: #ced4da;
}

.c4:active {
  background-color: transparent;
  color: #868e96;
}

<InputWithIconButton
  error={null}
  fontSize={1}
  icon={<div />}
  onAction={[Function]}
  type="text"
>
  <div>
    <Flex>
      <Box
        className="c0"
      >
        <div
          className="c0 c1"
        >
          <Box
            mt={2}
            w={0.9}
          >
            <div
              className="c2"
            >
              <Input
                error={null}
                fontSize={1}
                icon={<div />}
                onAction={[Function]}
                onChange={[Function]}
                rows={3}
                type="text"
              >
                <div>
                  <Input__StyledInput
                    error={null}
                    fontSize={1}
                    icon={<div />}
                    onAction={[Function]}
                    onChange={[Function]}
                    rows={3}
                    type="text"
                  >
                    <input
                      className="c3"
                      fontSize={1}
                      icon={<div />}
                      onChange={[Function]}
                      rows={3}
                      type="text"
                    />
                  </Input__StyledInput>
                </div>
              </Input>
            </div>
          </Box>
          <Box
            w={0.1}
          >
            <div
              className="c1"
            >
              <IconButton
                fontSize={3}
                icon={<div />}
                onClick={[Function]}
                pl={3}
                pr={3}
              >
                <IconButton__StyledIconButton
                  fontSize={3}
                  icon={<div />}
                  onClick={[Function]}
                  pl={3}
                  pr={3}
                >
                  <Button
                    border={2}
                    className="c4"
                    fontSize={3}
                    hollow={false}
                    icon={<div />}
                    inverted={false}
                    onClick={[Function]}
                    pl={3}
                    pr={3}
                  >
                    <Button__StyledButton
                      border={2}
                      className="c4 c5"
                      fontSize={3}
                      hollow={false}
                      icon={<div />}
                      inverted={false}
                      onClick={[Function]}
                      pl={3}
                      pr={3}
                    >
                      <button
                        className="c4 c5 c6"
                        fontSize={3}
                        icon={<div />}
                        onClick={[Function]}
                      >
                        <div />
                      </button>
                    </Button__StyledButton>
                  </Button>
                </IconButton__StyledIconButton>
              </IconButton>
            </div>
          </Box>
        </div>
      </Box>
    </Flex>
  </div>
</InputWithIconButton>
`;

exports[`InputWithIconButton renders 1`] = `
.c1 {
  box-sizing: border-box;
}

.c2 {
  box-sizing: border-box;
  margin-top: 8px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c6 {
  border-radius: 56px;
  display: inline-block;
  height: 70px;
  line-height: 1;
  padding-left: 56px;
  padding-right: 56px;
  text-transform: uppercase;
  vertical-align: middle;
  -webkit-tap-highlight-color: transparent;
  outline: 0;
  font-family: inherit;
  font-weight: 600;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #f8f9fa;
  background-color: #0588cb;
  border: none;
  text-align: center;
  -webkit-letter-spacing: .5px;
  -moz-letter-spacing: .5px;
  -ms-letter-spacing: .5px;
  letter-spacing: .5px;
  z-index: 1;
  -webkit-transition: .2s ease-out;
  transition: .2s ease-out;
  cursor: pointer;
}

.c6:hover {
  border: none;
  background-color: #37a6ed;
}

.c6:active {
  -webkit-transition: .2s ease-in;
  transition: .2s ease-in;
  border: none;
  background-color: #0f77b8;
}

.c5 {
  padding-left: 16px;
  padding-right: 16px;
  font-size: 28px;
}

.c3 {
  font-size: 16px;
  font-family: inherit;
  display: 'block';
  vertical-align: 'middle';
  background-color: transparent;
  border: none;
  border-bottom: 1px solid #f1f3f5;
  border-radius: 0;
  margin: 0;
  padding: 0;
  outline: none;
  width: 100%;
  height: 32px;
  box-shadow: none;
  box-sizing: content-box;
  -webkit-transition: all .3s;
  transition: all .3s;
}

.c3:hover:not([readonly]) {
  border-bottom: 1px solid #dee2e6;
}

.c3:focus:not([readonly]) {
  border-bottom: 2px solid #0588cb;
}

.c4 {
  padding-left: 16px;
  padding-right: 16px;
  font-size: 28px;
  border-radius: none;
  color: #adb5bd;
  background-color: transparent;
}

.c4:hover {
  background-color: transparent;
  color: #ced4da;
}

.c4:active {
  background-color: transparent;
  color: #868e96;
}

<InputWithIconButton
  error={null}
  fontSize={1}
  icon={<div />}
  type="text"
>
  <div>
    <Flex>
      <Box
        className="c0"
      >
        <div
          className="c0 c1"
        >
          <Box
            mt={2}
            w={0.9}
          >
            <div
              className="c2"
            >
              <Input
                error={null}
                fontSize={1}
                icon={<div />}
                onChange={[Function]}
                rows={3}
                type="text"
              >
                <div>
                  <Input__StyledInput
                    error={null}
                    fontSize={1}
                    icon={<div />}
                    onChange={[Function]}
                    rows={3}
                    type="text"
                  >
                    <input
                      className="c3"
                      fontSize={1}
                      icon={<div />}
                      onChange={[Function]}
                      rows={3}
                      type="text"
                    />
                  </Input__StyledInput>
                </div>
              </Input>
            </div>
          </Box>
          <Box
            w={0.1}
          >
            <div
              className="c1"
            >
              <IconButton
                fontSize={3}
                icon={<div />}
                pl={3}
                pr={3}
              >
                <IconButton__StyledIconButton
                  fontSize={3}
                  icon={<div />}
                  pl={3}
                  pr={3}
                >
                  <Button
                    border={2}
                    className="c4"
                    fontSize={3}
                    hollow={false}
                    icon={<div />}
                    inverted={false}
                    pl={3}
                    pr={3}
                  >
                    <Button__StyledButton
                      border={2}
                      className="c4 c5"
                      fontSize={3}
                      hollow={false}
                      icon={<div />}
                      inverted={false}
                      pl={3}
                      pr={3}
                    >
                      <button
                        className="c4 c5 c6"
                        fontSize={3}
                        icon={<div />}
                      >
                        <div />
                      </button>
                    </Button__StyledButton>
                  </Button>
                </IconButton__StyledIconButton>
              </IconButton>
            </div>
          </Box>
        </div>
      </Box>
    </Flex>
  </div>
</InputWithIconButton>
`;
